siphon/simplewebservice/igra2.py

```diff
@@ -102,15 +103,17 @@ def _get_data_raw(self):
         Returns a tuple with a string for the body, string for the headers,
         and a list of dates.
         """
-        # Import need to be here so we can monkeypatch urlopen for testing and avoid
-        # downloading live data for testing
-        try:
-            from urllib.request import urlopen
-        except ImportError:
-            from urllib2 import urlopen
-
-        with closing(urlopen(self.ftpsite + self.site_id + self.suffix + '.zip')) as url:
-            f = ZipFile(BytesIO(url.read()), 'r').open(self.site_id + self.suffix)
+        path = self.folder + self.site_id + self.suffix + '.zip'
+
+        resp = self.get_path(path)
+        # See if the return is valid, but has no data
+        if resp.raise_for_status() is not None:
```

Comment:
Not quite right. So if you look in the catalog code, we do:

https://github.com/Unidata/siphon/blob/b3c6f2de31702d3f00c3b8827f6278c039494144/siphon/catalog.py#L274-L281

What happens is that if an HTTP error was returned, it will raise an error, otherwise things continue fine. (See [here](https://2.python-requests.org/en/master/user/quickstart/#response-status-codes) for the docs.) So when you call this function, if there's an error, you'll never get to the code to get the ValueError. I think that `ValueError` with that message is more helpful to our users, so we should keep it. You'll need to wrap `raise_for_status()` in a `try...except` to handle `http_util.HTTPError`--then the exception handling can raise the new `ValueError`.

Reply:
Ah, ok that makes more sense. I'll fix that.

Reply:
Ok, let me know if that's better or not